pkg/backingstore/backingstore.go

```diff
@@ -350,9 +350,17 @@ func createCommon(cmd *cobra.Command, args []string, storeType nbv1.StoreType, p
 	o = util.KubeObject(bundle.File_deploy_internal_secret_empty_yaml)
 	secret := o.(*corev1.Secret)
 	secret.Namespace = options.Namespace
-	secret.Name = fmt.Sprintf("backing-store-%s-%s", storeType, name)
-	secret.StringData = map[string]string{}
-	secret.Data = nil
+	if secretName != "" {
+		secret.Name = secretName
+		if !util.KubeCheck(secret) {
```

Comment:
Why do we use negative conditions?  
I find it confusing (In General) and I think we should avoid negative conditions as much as we can. 
WDYT?

Reply:
How about that?
- https://github.com/noobaa/noobaa-operator/blob/24022e21a3881fb64d2ddf1eccf799eb4c6c4f01/pkg/backingstore/backingstore.go#L409

I don't mind either way.

Reply:
I meant something like:
```suggestion
		if util.KubeCheck(secret) {
```
and straightening the logic:
```suggestion
		if util.KubeCheck(secret) {
	secret.Name = fmt.Sprintf("backing-store-%s-%s", storeType, name)
		secret.StringData = map[string]string{}
		secret.Data = nil
	} else {
			log.Fatalf(`❌ Could not find the suggested secret: name %q namespace %q`, secret.Name, secret.Namespace)
			return
		}

	}
```
Besides that, as written in https://github.com/noobaa/noobaa-operator/pull/992#discussion_r1008887777 I dont think we should fail at all.